micro_manager/snapshot/snapshot.py

```diff
@@ -0,0 +1,281 @@
+        self._crashed_snapshots = []  # Declaration
+
+    # **************
+    # Public methods
+    # **************
+
+    def solve(self) -> None:
+        """
+        Solve the problem by iterating over a set macro parameters.
+        - Create micro simulation object.
+        - Post-process micro output.
+        - Write output to database.
+        - Merge output in parallel run.
+        """
+
+        # Loop over all macro parameters
+        for elems in range(self._local_number_of_sims):
+            # Create micro simulation object
+            self._micro_sims = create_simulation_class(self._micro_problem)(
+                self._global_ids_of_local_sims[elems]
+            )
+
+            micro_sims_input = self._macro_parameters[elems]
+            # Solve micro simulation
+            micro_sims_output = self._solve_micro_simulation(micro_sims_input)
+
+            # Write output to file
+            if micro_sims_output is not None:
+                # Post-processing
+                if self._post_processing_file_name is not None:
+                    # Attempt importing post-processing script
+                    try:
+                        post_processing = getattr(
+                            importlib.import_module(
+                                self._post_processing_file_name, "Postprocessing"
+                            ),
+                            "Postprocessing",
+                        )
+                        if hasattr(post_processing, "postprocessing") and callable(
+                            getattr(post_processing, "postprocessing")
+                        ):
+                            micro_sims_output = post_processing.postprocessing(
+                                micro_sims_output
+                            )
+                        else:
+                            self._logger.info(
+                                "No post-processing script with the provided path found. Skipping post-processing."
+                            )
+                            self._post_processing_file_name = None
+                    except Exception:
+                        self._logger.info(
+                            "No post-processing script with the provided path found. Skipping post-processing."
+                        )
+                        self._post_processing_file_name = None
+                self._data_storage.write_output_to_hdf(
+                    self._output_file_path,
+                    micro_sims_input,
+                    micro_sims_output,
+                    elems,
+                    len(self._macro_parameters),
+                )
+            # Log error and wrote macro data to database if simulation has crashed
+            else:
+                parameter = ""
+                for key, value in micro_sims_input.items():
+                    parameter += "{} = {}, ".format(key, value)
+                parameter = parameter[:-2]
```

Comment:
What and why is this?

Reply:
This logs the input parameters for which a crash occurred providing this information to the user immediately. I guess it can be removed as it is also stored elsewhere.